task_number/odd_filter.dart

```diff
@@ -0,0 +1,47 @@
+import 'filter.dart';
+
+class OddFilter extends Filter{
+  OddFilter(String numbers) : super (numbers);
+
+  @override
+  List filterNumbers(List numbers){
+    List oddNumbers = [];
+    for (int i = 0; i < numbers.length; i++) {
+      if (!isEven(numbers, i)) {
+        oddNumbers.add(numbers[i]);
+      }
+    }
+    return oddNumbers;
+  }
+
+  @override
+  int addUpNumbers() {
+    int number =0;
+    int sumOfOddNumbers = 0;
+    for (int i = 0; i < filterNumbers(convertStringToList()).length; i++) {
+      number = filterNumbers(convertStringToList())[i];
```

Comment:
Drugi raz wywołujesz tą samą metodę, w jednej funkcji, można wywołać raz i zapisać resultat do zmiennej, wtedy nie jest wykonywana 2 razy ta sama rzecz. Co więcej robisz to w pętli więc tak na prawdę wołasz ją wielokrotnie.